localstack/services/lambda_/event_source_listeners/adapters.py

```diff
@@ -74,52 +74,79 @@ def __init__(self, lambda_service: LambdaService):
         self.lambda_service = lambda_service
 
     def invoke(self, function_arn, context, payload, invocation_type, callback=None):
+        request_id = gen_amzn_requestid()
+        self._invoke_async(request_id, function_arn, context, payload, invocation_type, callback)
+
+    def _invoke_async(
```

Comment:
We should not count this as "async invoke" because that is purely our internal legacy construct of dispatching invokes. Conceptually, it is just a single synchronous invoke that happens here!

Reply:
understand, thanks!